components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -88,7 +89,27 @@ public void handle(HttpServletRequest request, HttpServletResponse response,
 
         String fidp = request.getParameter(FrameworkConstants.RequestParams.FEDERATED_IDP);
 
-        Map<String, AuthenticatedIdPData> authenticatedIdPs = context.getPreviousAuthenticatedIdPs();
+        Map<String, AuthenticatedIdPData> authenticatedIdPs = context.getCurrentAuthenticatedIdPs();
+
+        // If there are no current authenticated IDPs, it means no authentication has been taken place yet.
+        // So see whether there are previously authenticated IDPs for this session.
+        // NOTE : currentAuthenticatedIdPs (if not null) always contains the previousAuthenticatedIdPs
+        if (MapUtils.isEmpty(authenticatedIdPs)) {
+            authenticatedIdPs = context.getPreviousAuthenticatedIdPs();
+            if (log.isDebugEnabled()) {
+                log.debug("No current authenticated IDPs in the authentication context. " +
+                        "Continuing with the previous authenticated IDPs");
+            }
+        }
+
+        if (log.isDebugEnabled()) {
+            if (authenticatedIdPs == null) {
+                log.debug("No previous or current authenticated IDPs found in the authentication context. " +
+                        "Both are null");
+            } else {
+                log.debug(String.format("%d  authenticated IDPs found.", authenticatedIdPs.size()));
```

Comment:
It will look odd when a line like
131 authenticated IDPs found.

Can we change it to 
"Found authenticated IdPs. Count:  %d"